filters/united-states/new-york/syracuse/public-art-geojson-to-json.js

```diff
@@ -1,40 +1,99 @@
+  
+    // convert JSON data to object form
     if (typeof data === 'string' || data instanceof String) {
         data = JSON.parse(data);
     }
-    
+  
+    // define new data and errors array
     let new_data = [];
-
+    let errors = [];
+  
+    // iterate through each data entry
     data.features.map(d => {
         let item = {};
-        item.name = d.properties.Title;
-        if (item.name === undefined) {
-            console.log(`Data name not found for art with url ${d.url}`);
+        let skip = false;
+  
+        // add name (required)
+        if (!add_required(item, "name", d, d.properties.Title, errors)) {
+            skip = true;
+        }
+  
+        // add coordinates (required)
+        let coordinates = {};
+        if (!add_required(coordinates, "longitude", d, d.properties.Longitude, errors)) {
+            skip = true;
         }
-        let coordinates = { longitude: parseInt(d.properties.Longitude), latitude: d.properties.Latitude};
-        if (coordinates.longitude === undefined || coordinates.latitude === undefined) {
-            console.log(`Data coordinates not found for art with url ${d.url}`);
+        if (!add_required(coordinates, "latitude", d, d.properties.Latitude, errors)) {
+            skip = true;
         }
         item.coordinates = coordinates;
-
-        let details = {};
-        details.type = d.properties.Type;
-        details.address = d.properties.Address;
-        details.artist_first_name = d.properties.Artist_First;
-        details.artist_last_name = d.properties.Artist_Last_;
-        details.media = String(d.properties.Media);
-        details.year_created = String(d.properties.Year_Created);
-        details.neighbourhood = d.properties.Neighborhood;
-        details.image_url = String(d.properties.Image_Url);
-
-        item.details = details;
-        new_data.push(item);
+        
+        let full_name = ""
+        if(d.properties.Artist_First != null && d.properties.Artist_Last_ != null){
+          full_name = d.properties.Artist_First + " " + d.properties.Artist_Last_
+        }else if(d.properties.Artist_First != null){
+          full_name = d.properties.Artist_First
+        }
+        add_if_not_null(item, "artist", full_name);
+        
+        item.dates = create_dates_template();
+        add_if_not_null(item.dates.created, "year", d.properties.Year_Created);
+        remove_null_date_fields(item)
+  
+        item.address={};
+        add_if_not_null(item.address, "street_address", d.properties.Address);
+  
+        add_if_not_null(item, "material", d.properties.Media);
+        add_if_not_null(item, "type", d.properties.Type);
+        add_if_not_null(item, "area", d.properties.Neighborhood);
+  
+        item.image_urls = [];
+        image_urls = [d.properties.Image_Url];
+        for (let i = 0; i < image_urls.length; i++) {
+          if (image_urls[i] != null) {
+            item.image_urls.push(image_urls[i])
+          }
+        }
```

Comment:
again, check that there are items in item.image_urls, and if its an empty array, remove this field from item